core/domain/translatable_object_registry.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Registry for translatable objects."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import inspect
+
+from extensions.objects.models import objects
+import python_utils
+
+
+class Registry(python_utils.OBJECT):
+    """Registry of all translatable objects."""
+
+    # Dict mapping object class names to their classes.
+    translatable_objects_dict = {}
+
+    @classmethod
+    def _refresh_registry(cls):
+        """Refreshes the registry by adding new translatable object classes
+        to the registry.
+        """
+        cls.translatable_objects_dict.clear()
+
+        # Add new object instances to the registry.
+        for name, clazz in inspect.getmembers(
+                objects, predicate=inspect.isclass):
+            if name.endswith('_test') or name.startswith('Base'):
+                continue
+
+            ancestor_names = [
+                base_class.__name__ for base_class in inspect.getmro(clazz)]
+            # Some classes, such as TranslatableHtmlContentId, are not
+            # subclasses of BaseTranslatableObject, despite starting with the
+            # string 'Translatable'. So we need to do verification based on the
+            # class's ancestors.
+            if 'BaseTranslatableObject' not in ancestor_names:
+                continue
+            cls.translatable_objects_dict[clazz.__name__] = clazz
+
+    @classmethod
+    def get_all_class_names(cls):
+        """Gets a list of all translatable object class names.
+
+        Returns:
+            list(str). The full sorted list of translatable object class names.
+        """
+        cls._refresh_registry()
+        return sorted(cls.translatable_objects_dict.keys())
+
+    @classmethod
+    def get_object_class(cls, obj_type):
```

Comment:
I think the current name is fine, since we use it in multiple places throughout the codebase, so I'm leaving it as-is so as to keep consistency. I think the "type" terminology comes from the fact that this is the string you get when you do `type(object)` in Python.